package/src/Guard.tsx

```diff
@@ -133,8 +133,17 @@ const Guard: React.FunctionComponent<GuardProps> = ({ children, component, meta,
       const { props, redirect } = await resolveAllGuards();
       pageProps = props;
       routeRedirect = redirect;
+
+      // If there's a redirect, determine if it'll be infinite and throw error if so
+      if (routeRedirect) {
+        const redirectPath =
+          typeof routeRedirect === 'string' ? routeRedirect : routeRedirect.pathname;
```

Comment:
```suggestion
        const redirectPath = routeRedirect?.pathname || routeRedirect;
```

Reply:
Oh that's actually a TypeScript thing! `routeRedirect` can either be an object or string, so we need to type-check it first before accessing the `pathname` property

There's also this way to write it if it's preferable!:
```tsx
let redirectPath: string | undefined;
if (typeof routeRedirect === 'string') {
  redirectPath = routeRedirect;
} else {
  redirectPath = routeRedirect.pathname;
}
```